src/lsm/manifest_level.zig

```diff
@@ -37,7 +38,39 @@ pub fn ManifestLevelType(
             .{},
         );
 
-        pub const Tables = SegmentedArray(TableInfo, NodePool, table_count_max, .{});
+        pub const Tables = SortedSegmentedArray(
+            TableInfo,
+            NodePool,
+            table_count_max,
+            KeyAndSnapshot,
+            struct {
+                inline fn key_from_value(table_info: *const TableInfo) KeyAndSnapshot {
+                    return .{
+                        .key_max = table_info.key_max,
+                        .snapshot_min = table_info.snapshot_min,
+                    };
+                }
+            }.key_from_value,
+            struct {
+                inline fn compare_key_and_snapshot(
+                    key_a: KeyAndSnapshot,
```

Comment:
I'll add a TODO to `SegmentedArray` to benchmark it. Based on [this talk](https://www.youtube.com/watch?v=dEIsJPpCZYg) I think there's a good chance that since the function is inline the copy is being optimized away anyhow.

Reply:
Yes, I'm wondering more _how_ can we change the interface to make the `*const` explicit, to move in that direction, not to detect whether the copy is happening or not.

My thinking on this is that we actually want to tell the compiler that we don't need a copy. Often, it can figure that out, but better to be explicit to help it out.

Reply:
:thinking: Don't `key_a: KeyAndSnapshot` and `key_a: *const KeyAndSnapshot` both tell the compiler that we don't need a copy? i.e. (since `KeyAndSnapshot` has no pointers) both of those make it clear that the function does not mutate `key_a`'s memory, so it is free to use a pointer of a copy (whichever the optimizer decides is better).

Reply:
> The latter explicitly says we _must_

This sounds like a drawback to me -- for small keys (such as `KeyAndSnapshot`) copying may well be faster than the indirection of a pointer.

Reply:
> This sounds like a drawback to me -- for small keys (such as KeyAndSnapshot) copying may well be faster than the indirection of a pointer.

Touché, agreed.

Thanks @batiati also for the clarification on inline functions.

This is like a bugbear for me that keeps coming up. 😛